flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/scheduler/Scheduler.java

```diff
@@ -152,12 +150,14 @@ else if (ret instanceof Future) {
 			}
 		}
 		catch (NoResourceAvailableException e) {
-			return FlinkCompletableFuture.completedExceptionally(e);
+			CompletableFuture<SimpleSlot> notEnoughResources = new CompletableFuture<>();
```

Comment:
How about introducing a utility method to create an exceptionally completed `CompletableFuture` in one line?